backend/src/planner/schemas.ts

```
@@ -0,0 +1,128 @@
+/**
+ * Schema for PlannerCharacter
+ */
+export const PlannerCharacterSchema = z.object({
+  name: z.string().min(1, 'Character name is required'),
+  description: z.string().min(1, 'Character description is required'),
+  asset_filename: z
+    .string()
+    .min(1, 'Asset filename is required')
+    .regex(/^[a-z0-9_-]+\.png$/, 'Asset filename must be kebab-case or snake_case with .png extension')
+    .refine(
+      (filename) => /^[\x00-\x7F]+$/.test(filename),
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)